pkg/controllers/hypershiftdeployment_controller.go

```diff
@@ -141,7 +141,19 @@ func (r *HypershiftDeploymentReconciler) Reconcile(ctx context.Context, req ctrl
 
 	if configureInfra {
 		if hyd.Spec.Infrastructure.Platform == nil {
-			return ctrl.Result{}, r.updateMissingInfrastructureParameterCondition(&hyd, "Missing value HypershiftDeployment.Spec.Infrastructure.Platform")
+			return ctrl.Result{},
+				r.updateMissingInfrastructureParameterCondition(&hyd,
+					"Missing value HypershiftDeployment.Spec.Infrastructure.Platform")
+		}
+
+		err = r.annotateManagedClusterVersion(ctx, &hyd)
```

Comment:
This is the new method that looks up the Hosting Service Cluster version and writes the annotation.